### PR TITLE
Always process embedded messages

### PR DESCRIPTION
**Pull request by sergefdrv**
This pull request prepares replica message processing for view change implementation. It resolves issues related to processing indirectly received messages and requests propagated from former views.

### CODE REVIEW

File: core/message-handling_test.go [hunk 1]
```diff
@@ -388,35 +411,18 @@ func TestMakeApplicableReplicaMessageProcessor(t *testing.T) {
...
-	}
 	applyReplicaMessage := func(msg messages.ReplicaMessage) error {
 		args := mock.MethodCalled("replicaMessageApplier", msg)
 		return args.Error(0)
 	}
-	process := makeApplicableReplicaMessageProcessor(processMessage, applyReplicaMessage)
+	process := makeApplicableReplicaMessageProcessor(applyReplicaMessage)
 
 	msg := mock_messages.NewMockReplicaMessage(ctrl)
-	embeddedMsgs := []interface{}{
```
Comment (ynamiki) [inline core/message-handling_test.go:402]: By moving this part to `TestMakeReplicaMessageProcessor()`, an element of `embeddedMsgs` has been changed to an empty struct. It is my understanding that there is no one who see the content so there are no difference in what is tested by the codes.

**Review by ynamiki — APPROVED** [review]

File: core/message-handling_test.go (hunk 1 above)
Comment (sergefdrv) [reply core/message-handling_test.go:402]: You are right, `testify/mock` uses `reflect.DeepEqual` to compare expected arguments which will consider empty structures as equal. Thanks for spotting this!

File: core/message-handling_test.go [hunk 2]
```diff
@@ -259,6 +270,18 @@ func TestMakeReplicaMessageProcessor(t *testing.T) {
 
 	uiMsg.EXPECT().ReplicaID().Return(otherID).AnyTimes()
 
+	mock.On("messageProcessor", embeddedMsgs[0]).Return(false, fmt.Errorf("Error")).Once()
+	_, err = process(uiMsg)
+	assert.Error(t, err, "Failed to process embedded message")
+
+	mock.On("messageProcessor", embeddedMsgs[0]).Return(true, nil).Once()
+	mock.On("messageProcessor", embeddedMsgs[0]).Return(false, fmt.Errorf("Error")).Once()
```
Comment (sergefdrv) [inline core/message-handling_test.go]: Just noticed that the index is repeated. Will fix this as well

**Comment by sergefdrv** [comment]
Summary of [changes](https://github.com/hyperledger-labs/minbft/compare/261f7fbb9c8c1b20970ae035e3378f614c086598..178bea609f124c5c6f102a5865495674548b209d):

- Fixed `TestMakeReplicaMessageProcessor`

**Comment by sergefdrv** [comment]
Summary of [changes](https://github.com/hyperledger-labs/minbft/compare/178bea609f124c5c6f102a5865495674548b209d..1ab62465de49f49391ac5ef51ab7343b28cb86a9):

- Rebase

**Review by nhoriguchi — APPROVED** [review]

**Pull request merged**